### PR TITLE
Mention in docs what determines a succeeded pipeline

### PR DESCRIPTION
**Pull request by Hi-Angel**
While on it, also improve exception reporting for cases, where there're stdout and stderr set that should be printed, but they don't at the moment.

Fixes: #281

### CODE REVIEW

File: marge.app [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
+from traceback import print_exc
+
+try:
+    marge.app.main()
+except Exception as e:
+    print_exc()
+    if hasattr(e, 'stdout'):
+        print(f'stdout was: {e.stdout}')
+    if hasattr(e, 'stderr'):
+        print(f'stderr was: {e.stderr}')
```
Comment (tclh123) [inline marge.app:12]: Hey, could you give examples for this?
What did you see before and after this change for your case?

And maybe this change could be
```
try:
    marge.app.main()
except Exception as e:
    if hasattr(e, 'stdout'):
        print(f'stdout was: {e.stdout}')
    if hasattr(e, 'stderr'):
        print(f'stderr was: {e.stderr}')
    raise
```

**Review by tclh123 — CHANGES_REQUESTED** [review]

**Comment by tclh123** [comment]
Thanks for you contribution!

File: marge.app (hunk 1 above)
Comment (Hi-Angel) [reply marge.app:12]: Sure. Before:

```
 λ ./marge.app --config-file=~/.config/marge-bot.yml
2020-12-18 16:21:18,544 INFO Finding out my current projects...
2020-12-18 16:21:20,052 INFO Fetching merge requests assigned to me in K.Kharlamov/bds...
2020-12-18 16:21:20,282 INFO Running git clone --origin=origin git@gitlab.npobaum.ru:K.Kharlamov/bds.git /tmp/tmp8y0axpy0/tmpwmu6975g
Traceback (most recent call last):
  File "/home/constantine/Projects/marge-bot/marge/git.py", line 198, in _run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/usr/lib/python3.9/subprocess.py", line 1130, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "/usr/lib/python3.9/subprocess.py", line 1978, in _communicate
    self._check_timeout(endtime, orig_timeout, stdout, stderr)
  File "/usr/lib/python3.9/subprocess.py", line 1174, in _check_timeout
    raise TimeoutExpired(
subprocess.TimeoutExpired: Command '[b'git', b'clone', b'--origin=origin', b'git@gitlab.npobaum.ru:K.Kharlamov/bds.git', b'/tmp/tmp8y0axpy0/tmpwmu6975g']' timed out after 1 seconds

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/constantine/Projects/marge-bot/./marge.app", line 3, in <module>
    marge.app.main()
  File "/home/constantine/Projects/marge-bot/marge/app.py", line 333, in main
    marge_bot.start()
  File "/home/constantine/Projects/marge-bot/marge/bot.py", line 42, in start
    self._run(repo_manager)
  File "/home/constantine/Projects/marge-bot/marge/bot.py", line 57, in _run
    self._process_projects(
  File "/home/constantine/Projects/marge-bot/marge/bot.py", line 98, in _process_projects
    self._process_merge_requests(repo_manager, project, merge_requests)
  File "/home/constantine/Projects/marge-bot/marge/bot.py", line 143, in _process_merge_requests
    repo = repo_manager.repo_for_project(project)
  File "/home/constantine/Projects/marge-bot/marge/store.py", line 24, in repo_for_project
    repo.clone()
  File "/home/constantine/Projects/marge-bot/marge/git.py", line 36, in clone
    self.git('clone', '--origin=origin', reference_flag, self.remote_url,
  File "/home/constantine/Projects/marge-bot/marge/git.py", line 185, in git
    return _run(*command, env=env, check=True, timeout=timeout_seconds)
  File "/home/constantine/Projects/marge-bot/marge/git.py", line 202, in _run
    raise TimeoutExpired(
subprocess.TimeoutExpired: Command '[b'git', b'clone', b'--origin=origin', b'git@gitlab.npobaum.ru:K.Kharlamov/bds.git', b'/tmp/tmp8y0axpy0/tmpwmu6975g']' timed out after 1 seconds
```

After:

```
 λ ./marge.app --config-file=~/.config/marge-bot.yml
2020-12-18 16:21:38,490 INFO Finding out my current projects...
2020-12-18 16:21:39,290 INFO Fetching merge requests assigned to me in K.Kharlamov/bds...
2020-12-18 16:21:39,615 INFO Running git clone --origin=origin git@gitlab.npobaum.ru:K.Kharlamov/bds.git /tmp/tmp8fuyqpz3/tmp8g4r1vin
Traceback (most recent call last):
  File "/home/constantine/Projects/marge-bot/marge/git.py", line 198, in _run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/usr/lib/python3.9/subprocess.py", line 1130, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "/usr/lib/python3.9/subprocess.py", line 1978, in _communicate
    self._check_timeout(endtime, orig_timeout, stdout, stderr)
  File "/usr/lib/python3.9/subprocess.py", line 1174, in _check_timeout
    raise TimeoutExpired(
subprocess.TimeoutExpired: Command '[b'git', b'clone', b'--origin=origin', b'git@gitlab.npobaum.ru:K.Kharlamov/bds.git', b'/tmp/tmp8fuyqpz3/tmp8g4r1vin']' timed out after 1 seconds

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/constantine/Projects/marge-bot/./marge.app", line 6, in <module>
    marge.app.main()
  File "/home/constantine/Projects/marge-bot/marge/app.py", line 333, in main
    marge_bot.start()
  File "/home/constantine/Projects/marge-bot/marge/bot.py", line 42, in start
    self._run(repo_manager)
  File "/home/constantine/Projects/marge-bot/marge/bot.py", line 57, in _run
    self._process_projects(
  File "/home/constantine/Projects/marge-bot/marge/bot.py", line 98, in _process_projects
    self._process_merge_requests(repo_manager, project, merge_requests)
  File "/home/constantine/Projects/marge-bot/marge/bot.py", line 143, in _process_merge_requests
    repo = repo_manager.repo_for_project(project)
  File "/home/constantine/Projects/marge-bot/marge/store.py", line 24, in repo_for_project
    repo.clone()
  File "/home/constantine/Projects/marge-bot/marge/git.py", line 36, in clone
    self.git('clone', '--origin=origin', reference_flag, self.remote_url,
  File "/home/constantine/Projects/marge-bot/marge/git.py", line 185, in git
    return _run(*command, env=env, check=True, timeout=timeout_seconds)
  File "/home/constantine/Projects/marge-bot/marge/git.py", line 202, in _run
    raise TimeoutExpired(
subprocess.TimeoutExpired: Command '[b'git', b'clone', b'--origin=origin', b'git@gitlab.npobaum.ru:K.Kharlamov/bds.git', b'/tmp/tmp8fuyqpz3/tmp8g4r1vin']' timed out after 1 seconds
stdout was: b''
stderr was: b"Cloning into '/tmp/tmp8fuyqpz3/tmp8g4r1vin'...\nfatal: early EOF\n"
```

Do you want me to add this to commit message?

Comment (Hi-Angel) [reply marge.app:12]: > And maybe this change could be

I'm not sure: the thing is: in this case Exception would be before the stdout/stderr output. So may not be obvious, what is this stdout/stderr about. Although… Maybe I could add a print "Exception happened", and then as you say?

Comment (Hi-Angel) [reply marge.app:11]: > Maybe I could add a print "Exception happened", and then as you say?

￼Done this, please see if you're okay with the result

It looks now like

```
 λ ./marge.app --config-file=~/.config/marge-bot.yml
2020-12-18 16:25:14,463 INFO Finding out my current projects...
2020-12-18 16:25:15,285 INFO Fetching merge requests assigned to me in K.Kharlamov/bds...
2020-12-18 16:25:15,678 INFO Running git clone --origin=origin git@gitlab.npobaum.ru:K.Kharlamov/bds.git /tmp/tmpw2mfe1ys/tmpab9sh1s0
Exception occured
stdout was: b''
stderr was: b"Cloning into '/tmp/tmpw2mfe1ys/tmpab9sh1s0'...\nfatal: early EOF\n"
Traceback (most recent call last):
  File "/home/constantine/Projects/marge-bot/marge/git.py", line 198, in _run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/usr/lib/python3.9/subprocess.py", line 1130, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "/usr/lib/python3.9/subprocess.py", line 1978, in _communicate
    self._check_timeout(endtime, orig_timeout, stdout, stderr)
  File "/usr/lib/python3.9/subprocess.py", line 1174, in _check_timeout
    raise TimeoutExpired(
subprocess.TimeoutExpired: Command '[b'git', b'clone', b'--origin=origin', b'git@gitlab.npobaum.ru:K.Kharlamov/bds.git', b'/tmp/tmpw2mfe1ys/tmpab9sh1s0']' timed out after 1 seconds

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/constantine/Projects/marge-bot/./marge.app", line 5, in <module>
    marge.app.main()
  File "/home/constantine/Projects/marge-bot/marge/app.py", line 333, in main
    marge_bot.start()
  File "/home/constantine/Projects/marge-bot/marge/bot.py", line 42, in start
    self._run(repo_manager)
  File "/home/constantine/Projects/marge-bot/marge/bot.py", line 57, in _run
    self._process_projects(
  File "/home/constantine/Projects/marge-bot/marge/bot.py", line 98, in _process_projects
    self._process_merge_requests(repo_manager, project, merge_requests)
  File "/home/constantine/Projects/marge-bot/marge/bot.py", line 143, in _process_merge_requests
    repo = repo_manager.repo_for_project(project)
  File "/home/constantine/Projects/marge-bot/marge/store.py", line 24, in repo_for_project
    repo.clone()
  File "/home/constantine/Projects/marge-bot/marge/git.py", line 36, in clone
    self.git('clone', '--origin=origin', reference_flag, self.remote_url,
  File "/home/constantine/Projects/marge-bot/marge/git.py", line 185, in git
    return _run(*command, env=env, check=True, timeout=timeout_seconds)
  File "/home/constantine/Projects/marge-bot/marge/git.py", line 202, in _run
    raise TimeoutExpired(
subprocess.TimeoutExpired: Command '[b'git', b'clone', b'--origin=origin', b'git@gitlab.npobaum.ru:K.Kharlamov/bds.git', b'/tmp/tmpw2mfe1ys/tmpab9sh1s0']' timed out after 1 seconds
```

**Comment by Hi-Angel** [comment]
While at it, let me change commit message real quick…

**Comment by Hi-Angel** [comment]
Okay, done.

**Review by tclh123 — APPROVED** [review]

**Comment by Hi-Angel** [comment]
Oops, not sure if it's intended, but in the repo the commits appear as squashed into one.

**Pull request merged**